crates/rust-client/src/lib.rs

```diff
@@ -255,16 +257,14 @@ impl Client {
     ///
     /// - `api`: An instance of [`NodeRpcClient`] which provides a way for the client to connect to
     ///   the Miden node.
+    /// - `rng`: An instance of [`FeltRng`] which provides randomness tools for generating new keys,
+    ///   serial numbers, etc. This can be any RNG that implements the [`FeltRng`] trait.
     /// - `store`: An instance of [`Store`], which provides a way to write and read entities to
     ///   provide persistence.
-    /// - `executor_store`: An instance of [`Store`] that provides a way for [`TransactionExecutor`]
-    ///   to retrieve relevant inputs at the moment of transaction execution. It should be the same
-    ///   store as the one for `store`, but it doesn't have to be the **same instance**.
     /// - `authenticator`: Defines the transaction authenticator that will be used by the
     ///   transaction executor whenever a signature is requested from within the VM.
-    /// - `in_debug_mode`: Instantiates the transaction executor (and in turn, its compiler) in
-    ///   debug mode, which will enable debug logs for scripts compiled with this mode for easier
-    ///   MASM debugging.
+    /// - `exec_options`: Execution options for the transaction executor, which can be used to
+    ///   configure the execution environment, such as whether to enable debug mode.
```

Comment:
Feel free to disregard, your version is fine as well, but suggestion:
```suggestion
    /// - `exec_options`: Options that control the transaction executor’s runtime behaviour (e.g. debug mode).
```